src/linux/x86_64/kvm_cpu.rs

```diff
@@ -35,6 +36,34 @@ const KVM_32BIT_GAP_SIZE: usize = 1024 << 20;
 // 3 GiB, aka. 0xC000_0000
 const KVM_32BIT_GAP_START: usize = KVM_32BIT_MAX_MEM_SIZE - KVM_32BIT_GAP_SIZE;
 
+impl From<&MmapMemory> for Vec<kvm_userspace_memory_region> {
+	fn from(mem: &MmapMemory) -> Self {
+		let sz = std::cmp::min(mem.memory_size, KVM_32BIT_GAP_START);
+		let mut regions = vec![kvm_userspace_memory_region {
+			slot: 0,
+			flags: mem.flags,
+			memory_size: sz as u64,
+			guest_phys_addr: mem.guest_address.as_u64(),
+			userspace_addr: mem.host_address as u64,
+		}];
+
+		if mem.memory_size > KVM_32BIT_GAP_START + KVM_32BIT_GAP_SIZE {
+			regions.push(kvm_userspace_memory_region {
+				slot: 1,
+				flags: mem.flags,
+				memory_size: (mem.memory_size - KVM_32BIT_GAP_START - KVM_32BIT_GAP_SIZE) as u64,
+				guest_phys_addr: mem.guest_address.as_u64()
+					+ (KVM_32BIT_GAP_START + KVM_32BIT_GAP_SIZE) as u64,
+				userspace_addr: (mem.host_address as usize
```

Comment:
This part worries me a bit, but it was already this way originally, so this concern is orthogonal to this PR.